src/circuit/hash.rs

```diff
@@ -130,9 +95,14 @@ fn hash_node(circ: &impl HugrView, node: Node, state: &mut HashState) -> u64 {
     let op = circ.get_optype(node);
     let mut hasher = FxHasher64::default();
 
+    // Hash the node children
     if circ.children(node).count() > 0 {
-        panic!("Cannot hash container node {node:?}.");
+        let container: SiblingGraph =
+            SiblingGraph::try_new(circ, node).expect("Cannot hash container node");
```

Comment:
AFAICS SiblingGraph does not insist its root node is a container; if its just a NodeHandle for a non-container, you'll get back a SiblingGraph with just one node (the root). So IIUC the only case where `try_new` will return a failure here is if `node` is bad, so you might tweak this message (or just unwrap!)